src/Lsp/Connection.ml

```diff
@@ -0,0 +1,59 @@
+(* This file is part of DBL, released under MIT license.
+ * See LICENSE for details.
+ *)
+
+(** Base HTML-like protocol *)
+
+open In_channel
+open Out_channel
+
+exception Connection_error of string
+
+type headers =
+  { content_length: int option
+  ; content_type: string option
+  }
+
+let parse_header headers line =
```

Comment:
This sounds like something we could use a third party library for unless it's pretty custom. (could refactor later)

Reply:
I'm not a fan of dependencies and the whole file is like 60 lines (and does exactly what's needed), so I'm not sure how much a library would help.